src/mpi/request/test.c

```diff
@@ -121,6 +121,7 @@ int MPI_Test(MPI_Request * request, int *flag, MPI_Status * status)
     MPIR_ERRTEST_INITIALIZED_ORDIE();
 
     MPID_THREAD_CS_ENTER(GLOBAL, MPIR_THREAD_GLOBAL_ALLFUNC_MUTEX);
+    MPID_THREAD_CS_ENTER(VCI, MPIR_THREAD_VCI_GLOBAL_MUTEX);
```

Comment:
nitpick: this and the ch4 wait patch could probably be squashed

Reply:
OK.